parquet/tests/encryption/encryption_async.rs

```diff
@@ -493,8 +503,341 @@ async fn read_and_roundtrip_to_encrypted_file_async(
+#[tokio::test]
+async fn test_concurrent_encrypted_writing_over_multiple_row_groups() {
+    // Read example data and set up encryption/decryption properties
+    let testdata = arrow::util::test_util::parquet_test_data();
+    let path = format!("{testdata}/encrypt_columns_and_footer.parquet.encrypted");
+    let file = std::fs::File::open(path).unwrap();
+
+    let file_encryption_properties = FileEncryptionProperties::builder(b"0123456789012345".into())
+        .with_column_key("double_field", b"1234567890123450".into())
+        .with_column_key("float_field", b"1234567890123451".into())
+        .build()
+        .unwrap();
+    let decryption_properties = FileDecryptionProperties::builder(b"0123456789012345".into())
+        .with_column_key("double_field", b"1234567890123450".into())
+        .with_column_key("float_field", b"1234567890123451".into())
+        .build()
+        .unwrap();
+
+    let (record_batches, metadata) =
+        read_encrypted_file(&file, decryption_properties.clone()).unwrap();
+    let schema = metadata.schema();
+
+    // Create a channel to send RecordBatches to the writer and send row groups
+    let (record_batch_tx, data) = tokio::sync::mpsc::channel::<RecordBatch>(100);
+    let data_generator = tokio::spawn(async move {
+        for record_batch in record_batches {
+            record_batch_tx.send(record_batch).await.unwrap();
+        }
+    });
+
+    let props = Some(
+        WriterPropertiesBuilder::default()
+            .with_file_encryption_properties(file_encryption_properties)
+            .build(),
+    );
+
+    // Create a temporary file to write the encrypted data
+    let temp_file = tempfile::tempfile().unwrap();
+    let arrow_writer =
+        ArrowWriter::try_new(&temp_file, metadata.schema().clone(), props.clone()).unwrap();
+
+    let (writer, row_group_writer_factory) = arrow_writer.into_serialized_writer().unwrap();
+    let max_row_groups = 1;
+
+    let (serialize_tx, serialize_rx) =
+        tokio::sync::mpsc::channel::<JoinHandle<RBStreamSerializeResult>>(max_row_groups);
+
+    let launch_serialization_task = spawn_parquet_parallel_serialization_task(
+        row_group_writer_factory,
+        data,
+        serialize_tx,
+        schema.clone(),
+    );
+
+    let _file_metadata = concatenate_parallel_row_groups(writer, serialize_rx)
+        .await
+        .unwrap();
+
+    data_generator.await.unwrap();
+    launch_serialization_task.await.unwrap().unwrap();
+
+    // Check that the file was written correctly
+    let (read_record_batches, read_metadata) =
+        read_encrypted_file(&temp_file, decryption_properties.clone()).unwrap();
+
+    assert_eq!(read_metadata.metadata().file_metadata().num_rows(), 50);
+    verify_encryption_test_data(read_record_batches, read_metadata.metadata());
+}
+
 #[tokio::test]
-async fn test_multi_threaded_encrypted_writing() {
+async fn test_multi_threaded_encrypted_writing_replace_deprecated_api() {
```

Comment:
I think the `replace_deprecated_api` isn't necessary and will only confuse readers who don't know all the history.

```suggestion
async fn test_multi_threaded_encrypted_writing() {
```